src/fshtypes/FshCodeSystem.ts

```diff
@@ -11,20 +10,26 @@ export class FshCodeSystem extends FshEntity {
   id: string;
   title?: string;
   description?: string;
-  concepts: FshConcept[];
-  rules: CaretValueRule[];
+  rules: (ConceptRule | CaretValueRule | InsertRule)[];
 
   constructor(public name: string) {
     super();
     this.id = name;
-    this.concepts = [];
     this.rules = [];
   }
 
-  addConcept(newConcept: FshConcept) {
-    if (this.concepts.find(existingConcept => existingConcept.code == newConcept.code)) {
+  addConcept(newConcept: ConceptRule) {
+    if (
+      this.rules
+        .filter(rule => rule instanceof ConceptRule)
+        .find((existingConcept: ConceptRule) => existingConcept.code == newConcept.code)
```

Comment:
I think this implementation effectively results in two loops: one that loops over all rules and keeps only the ConceptRules, and then another that loops over all those rules trying to match on the code.

I think this would be more efficient as a single `find`:
```ts
this.rules.find(rule => rule instanceof ConceptRule && rule.code == newConcept.code)
```
(not sure, but you might need a cast in there).